### PR TITLE
Fixed eventhubs default proxy configuration bug

### PR DESCRIPTION
**Pull request by ki1729**
# Description

Fixed the same issue as #24230 in EventHubClientBuilder

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### CODE REVIEW

**Review by liukun-msft — APPROVED** [review]
LGMT. May add a changelog for the fix?

File: sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusClientBuilderTest.java [hunk 1]
```diff
@@ -386,16 +386,16 @@ public void testConnectionWithAzureSasCredential() {
 
     private static Stream<Arguments> getProxyConfigurations() {
         return Stream.of(
-            Arguments.of("http://localhost:8080", true),
-            Arguments.of("localhost:8080", true),
+            //Arguments.of("http://localhost:8080", true),
```
Comment (conniey) [inline sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusClientBuilderTest.java:389]: commented out?

**Review by conniey — COMMENTED** [review]
one comment about the commented out options

File: sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusClientBuilderTest.java (hunk 1 above)
Comment (ki1729) [reply sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusClientBuilderTest.java:389]: Shouldn't be. Will remove

**Pull request merged**